### ISSUE TITLE
Emigre from Juice - query renaming episodes

### ISSUE THREAD

**Issue by jimlad68** [gpodder/gpodder#1482]
I have finally decided to replace my very old trustly but abandoned JUICE (iPodder) PODCAST downloader, mainly because it now fails to download some new https files. For the present it has come down to 3 alternatives; gPodder (son of juice I believe), MusicBee and AIMP. All have there up/downsides, or at least I do not understnd them.

For gPodder, Although lacking in some of the "fancy" features of the others (e.g. flexible folder+filename output, lots sorting columns), gPodder is definitely an improvement on Juice and importantly is still simple and easy to "see where I am at" i.e. run it, kick it off, let it do its stuff, close it.

[] The extension "Rename episode after the download" is very useful, however it would be nice if using both "sortdate" and "podcast_title" for an option for the "podcast_title" to come first.
- Also it would to nice to be able to use it just on some podcast feeds, as many already have a good naming convention. 
- Alternatively, is it possible to have 2 instances of gPodder with different settings?

**Comment by auouymous** [comment]
> Alternatively, is it possible to have 2 instances of gPodder with different settings?

Yes and no. Linux will not run two instances **as the same user** because it uses dbus to find other instances and abort. Windows and Mac use a fake dbus that doesn't offer this feature, which can lead to two instances accessing the same database, but will work fine if you give them separate `GPODDER_HOME`'s. There are however plans to use a real dbus one day, so gpodder should probably detect the other instance's `GPODDER_HOME` and only abort if they are the same.


> both instances would try to use the same GPODDER_DOWNLOAD_DIR

Separate `GPODDER_HOME` directories will have separate `GPODDER_DOWNLOAD_DIR` inside of them. Multiple instances can't share the same `GPODDER_HOME` or the same `GPODDER_DOWNLOAD_DIR`.


> The extension "Rename episode after the download" is very useful, however it would be nice if using both "sortdate" and "podcast_title" for an option for the "podcast_title" to come first.

There is already code for syncing that produces a filename from a format string. The rename extension could easily use that with very few changes. I will look into implementing it.


> Also it would to nice to be able to use it just on some podcast feeds, as many already have a good naming convention

That would add a lot of complexity.


> Separately I have looked to rename files with the Podcast "Publish Date"

The rename extension's sortdate is publish date.

**Comment by jimlad68** [comment]
[auouymous](https://github.com/auouymous)
Thanks for those detailed comments:

> re Separate GPODDER_HOME directories will have separate GPODDER_DOWNLOAD_DIR inside of them. Multiple instances can't share the same GPODDER_HOME or the same GPODDER_DOWNLOAD_DIR.

I have already set the GPODDER_DOWNLOAD_DIR via System Properties and Environment variables 
as in
[https://gpodder.github.io/docs/user-manual.html#gpodder-home-folder-and-download-location](url) and [https://superuser.com/questions/949560/how-do-i-set-system-environment-variables-in-windows-10/949577#949577](url)
(I aim to do similar for GPODDER_HOME in my final setup), but how do I setup those variables for different instances, is there a way to force a variable for same program from different program start folders/exe? Or is there a way to start the same program twice with different GPODDER_HOME and GPODDER_DOWNLOAD_DIR.

**Comment by auouymous** [comment]
It should be possible to create two separate icons to launch gpodder, and each one should be able to set custom environment variables. On Linux we would do `GPODDER_HOME="pathA" GPODDER_DOWNLOAD_DIR="pathB" gpodder`. The icon properties on Windows might have a separate text entry for environment, I don't know.

**Comment by jimlad68** [comment]
Firstly I have added to the Topic name
[auouymous](https://github.com/auouymous) Again thanks for your insight, although I did not understand how to use it, it gave me some food for thought. After many false starts I found this *.bat worked (for extra instances change the 2 for 1 or 3 or whatever):
_@echo on
rem set (for testing as is the pause)
 set GPODDER_HOME=E:\zPOD\gPodr2
setx GPODDER_HOME E:\zPOD\gPodr2
pause 0
start "" "E:\zPortAps\PortableApps\PortableApps\gPodderPortable\App\gPodder\data\bin\gpodder.exe" "E:\zPortAps\PortableApps\PortableApps\gPodderPortable\App\gPodder\data\bin"
::exit_
== Notes: From what I can gather!!:
- My USER had admin rights, but there were no "carry on" prompts.
- flip/flop between 2 different GPODDER_HOMEs worked well, I even had 2 working side by side. I will probably not do that as I have no idea if there could be any contentions. I suppose if all settings etc are stored in GPODDER_HOME it might be OK.
- set works only for that command session
- setx makes it permanent but only after the batch job has ended.
- So as far as I can work out, you need set if you call gPodder from the batch job, or setx if gPodder called after the Batch has closed. Hence I used both, especially during testing.
- I do not know if an **_exit_** is needed?
- I decided just to change GPODDER_HOME and not GPODDER_DOWNLOAD_DIR, as in my testing I got them out of sync and made things very confusing!
- I may use the standard portable gPodder app rather than via PortableApps, that might also get rid of the annoying startup message re "download path may be too long...", perhaps there is a way to stop that message especially when a shorter GPODDER_HOME is set.
- This command worked from the powershell
_[Environment]::SetEnvironmentVariable("GPODDER_HOME", "E:\zPOD\gPodrHome1", "User")_
but I could not get it to work via a *.BAT job with this line
_powershell -Command [Environment]::SetEnvironmentVariable("GPODDER_HOME", "E:\zPOD\gPodrHome1", "User")_
with many red lines starting with:
_At line:1 char:39 [Environment]::SetEnvironmentVariable(GPODDER_HOME, E:\zPOD\gPodrHome..... Missing ')' in method call._

PLEASE ANYONE: CORRECT ANY OF MY SUPPOSITIONS AND/OR ADD EASIER METHODS.

If this is a "good" fix, I hope it has helped others, perhaps adding to the docs.
